src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -39,8 +43,36 @@ public override bool Execute()
                 SleepTimeInMilliseconds = s_DefaultSleepTime;
             }
 
-            return CleanupAgentsAsync().Result;
+            bool returnValue = true;
+
+            if(Report)
+            {
+                ReportDiskUsage();
+            }
+            if(Clean)
+            {
+                returnValue &= CleanupAgentsAsync().Result;
+            }
+            // If report and clean are both 'true', then report disk usage both before and after cleanup.
+            if(Report && Clean)
+            {
+                ReportDiskUsage();
```

Comment:
Might be good to add a message before this report saying it is the size after clean-up.